component/common/loki/http/server.go

```diff
@@ -0,0 +1,94 @@
+package http
```

Comment:
The `TargetServer` runs both GRPC and HTTP, so this package may need a rename?
AFAIU we will be using and documenting the GRPC endpoints too in the future.

Reply:
Yes, it's actually the intention of this PR to expose the grpc configs